packages/angular_devkit/build_angular/src/browser/index2.ts

```diff
@@ -260,27 +369,46 @@ export function buildWebpackBrowser(
         normalize(workspace.getProject(projectName).root),
       );
 
-      return runWebpack(config, context, { logging: loggingFn }).pipe(
-        concatMap(buildEvent => {
-          if (buildEvent.success && !options.watch && options.serviceWorker) {
-            return from(augmentAppWithServiceWorker(
-              host,
-              root,
-              projectRoot,
-              resolve(root, normalize(options.outputPath)),
-              options.baseHref || '/',
-              options.ngswConfigPath,
-            ).then(() => ({ success: true }), () => ({ success: false })));
-          } else {
-            return of(buildEvent);
-          }
-        }),
-        map(event => ({
-          ...event,
-          outputPath: config.output && config.output.path || '',
-        } as BrowserBuilderOutput)),
-        concatMap(output => outputFn ? outputFn(output) : of(output)),
-      );
+      return combineLatest(
+        configs.map((config, idx) => runWebpack(
+          config,
+          context,
+          { logging: loggingFn })),
+      )
+        .pipe(
+          switchMap(buildEvents => {
+            if (buildEvents.length === 2) {
+              // if we are doing differential loading, we need to write the index.html
+              // console.debug('writing index.html', buildEvent);
+              return from(writeIndexHtml(buildEvents, options, root, host)).pipe(
+                map(_ => buildEvents),
+              );
+            }
+
+            return of(buildEvents);
+          }),
+          map(buildEvents => mergeBuildResult(buildEvents)),
```

Comment:
Seeing the usages we might only need if it was success or not hence. Most likely we can get away with using
```
     map(buildEvents => {success: buildEvents.every(x => x.success)}),
```